### PR TITLE
Feature/scalatest utils

### PR DESCRIPTION
**Pull request by wajda**
Adds some Scalatest utils:

1. `ConditionalTestTags` - runs certain tests conditionally
```scala
  it should "test that new Spark feature" taggedAs ignoreIf(ver"$SPARK_VERSION" < ver"2.4") in  {
    ...
  }

  it should "test some DAO" taggedAs ignoreIf(!isDatabaseAvailable) in  {
    ...
  }
```

2. `ConsoleStubs` - stubs console IO
```scala
  captureStdOut(Console.out.print("foo")) should be("foo")
```

3. `SystemExitFixture` - intercepts `System.exit()` and asserts status
```scala
  captureExitStatus(System.exit(42)) should be(42)

  // OR

  assertingExitStatus(be > 0 and be < 5) {
      // run some code that calls System.exit(...)
  }
```

4. `WhitespaceNormalizations` - extends Scalatest DSL with some whitespace treatment methods
```scala
  (
    """
      {
        a: 111,
        b: {
          v: 42
        }
      }
    """
    should equal ("{ a: 111, b: { v: 42 } }")
    (after being trimmed and whitespaceNormalized)
  )
```

### CODE REVIEW

File: src/test/scala/za/co/absa/commons/scalatest/WhitespaceNormalizationsSpec.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  extends AnyFlatSpec
+    with Matchers
+    with WhitespaceNormalizations {
+
+  it should "normalize whitespaces" in {
+    "  foo   bar \n  42  \t\t  " should equal(" foo bar 42 ")(after being whiteSpaceNormalised)
+  }
+
+  it should "remove whitespaces" in {
+    "  foo   bar \n  42  \t\t  " should equal("foobar42")(after being whiteSpaceRemoved)
```
Comment (Zejnilovic) [inline src/test/scala/za/co/absa/commons/scalatest/WhitespaceNormalizationsSpec.scala]: How does it behave with other whitespaces? `\r\f\v` There are whitespaces, that regexp recognizes.

File: src/test/scala/za/co/absa/commons/scalatest/ConsoleStubsSpec.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  it should "capture stderr" in {
+    captureStdErr(Console.err.print("bar")) should be("bar")
+  }
+
+  it should "populate stdin" in withStdIn("qux") {
+    Console.in.readLine() should be("qux")
+  }
+
+}
```
Comment (Zejnilovic) [inline src/test/scala/za/co/absa/commons/scalatest/ConsoleStubsSpec.scala]: missing new line

File: src/test/scala/za/co/absa/commons/lang/ARMSpec.scala [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  }
+}
+
+object ARMTest {
+
+  trait TestResource {
+    def close(): Unit
+  }
+
+}
```
Comment (Zejnilovic) [inline src/test/scala/za/co/absa/commons/lang/ARMSpec.scala]: missing new line

File: src/main/scala/za/co/absa/commons/scalatest/ConditionalTestTags.scala [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+package za.co.absa.commons.scalatest
+
+import org.scalatest.{Ignore, Tag}
+
+object ConditionalTestTags {
+
+  def ignoreIf(condition: => Boolean): Tag = Tag(if (condition) classOf[Ignore].getName else "")
+
+}
```
Comment (Zejnilovic) [inline src/main/scala/za/co/absa/commons/scalatest/ConditionalTestTags.scala]: missing new line

File: pom.xml [hunk 5]
```diff
@@ -94,14 +95,28 @@
...
             <optional>true</optional>
         </dependency>
+
+        <!-- Test scope dependencies -->
+
+        <dependency>
+            <groupId>org.scalatestplus</groupId>
+            <artifactId>mockito-1-10_${scala.binary.version}</artifactId>
+            <version>3.1.0.0</version>
+        </dependency>
```
Comment (yruslan) [inline pom.xml]: This dependency is commented to be in the `test` scope, yet the dependency itself is in the default (`compile`) scope.

**Comment by yruslan** [comment]
Looks very cool!

But consider adding more documentation to the public methods/classes/objects with examples like ones in the description of this PR.

File: pom.xml (hunk 5 above)
Comment (wajda) [reply pom.xml]: good catch ! :)

Comment (wajda) [reply pom.xml]: Sometime comments in code aren't that useless :smile:

File: src/test/scala/za/co/absa/commons/scalatest/WhitespaceNormalizationsSpec.scala (hunk 1 above)
Comment (wajda) [reply src/test/scala/za/co/absa/commons/scalatest/WhitespaceNormalizationsSpec.scala]: it simply replaces all `\s+` sequences with a single space character. So all of those should be covered.
Will add more tests

**Pull request merged**